server/src/main/java/org/elasticsearch/script/field/LongField.java

```diff
@@ -82,11 +82,27 @@ public static BigIntegerField toBigIntegerField(LongField sourceField) {
 
             @Override
             public BigInteger getNonPrimitiveValue() {
-                return BigInteger.valueOf(values.getLongValue());
+                return toBigInteger(values.getNonPrimitiveValue());
+            }
+
+            @Override
+            public long getLongValue() {
+                return toBigInteger(values.getNonPrimitiveValue()).longValue();
```

Comment:
Why convert from a long, to a BigInteger, back to long?

Reply:
Going through all of the converters and removing unnecessary additional conversions.